server/web/test/index.js

```diff
@@ -87,6 +88,39 @@ exports.register = function (server, options, next) {
     }
   });
 
+  // allows owner or admin to publish a test making it visible to the public
+  server.route({
+    method: 'GET',
+    path: '/{testSlug}/{rev}/publish',
+    handler: function (request, reply) {
+      pagesService.getBySlug(request.params.testSlug, request.params.rev)
```

Comment:
you should create an other method than **getBySlug** which start four queries behind. One is fine.